content/en/docs/tasks/configure-pod-container/resize-container-resources.md

```diff
@@ -61,205 +31,166 @@ maximum of a container's desired resource requests, and it's actual requests rep
+    * `reason: Deferred`: The requested resize is currently not possible, but might become feasible later (for example if another pod is removed). The Kubelet will retry the resize.
+    
 
-{{< note >}}
-If the Pod's `restartPolicy` is `Never`, container's resize restart policy must be
-set to `NotRequired` for all Containers in the Pod.
-{{< /note >}}
+* `type: PodResizeInProgress`: The Kubelet has accepted the resize and allocated resources, but the changes are still being applied. This is usually brief but might take longer depending on the resource type and runtime behavior. Any errors during actuation are reported in the `message` field (along with `reason: Error`).
 
-Below example shows a Pod whose Container's CPU can be resized without restart, but
-resizing memory requires the container to be restarted.
+## Container resize policies
+
+You can control whether a container should be restarted when resizing by setting `resizePolicy` within the container specification. This allows fine-grained control based on resource type (CPU or memory).
 
 ```yaml
-apiVersion: v1
-kind: Pod
-metadata:
-  name: qos-demo-5
-  namespace: qos-example
-spec:
-  containers:
-  - name: qos-demo-ctr-5
-    image: nginx
     resizePolicy:
     - resourceName: cpu
       restartPolicy: NotRequired
     - resourceName: memory
       restartPolicy: RestartContainer
-    resources:
-      limits:
-        memory: "200Mi"
-        cpu: "700m"
-      requests:
-        memory: "200Mi"
-        cpu: "700m"
 ```
 
+* `NotRequired`: (Default) Apply the resource change to the running container without restarting it.
+* `RestartContainer`: Restart the container to apply the new resource values. This is often necessary for memory changes because many applications and runtimes cannot adjust their memory allocation dynamically.
+
+If `resizePolicy[*].restartPolicy` is not specified for a resource, it defaults to `NotRequired`.
+
 {{< note >}}
-In the above example, if desired requests or limits for both CPU _and_ memory
-have changed, the container will be restarted in order to resize its memory.
+If a Pod's overall `restartPolicy` is `Never`, then any container `resizePolicy` must be `NotRequired` for all resources. You cannot configure a resize policy that would require a restart in such Pods.
 {{< /note >}}
 
-<!-- steps -->
+**Example Scenario:**
+
+Consider a container configured with `restartPolicy: NotRequired` for CPU and `restartPolicy: RestartContainer` for memory.
+* If only CPU resources are changed, the container is resized in-place.
+* If only memory resources are changed, the container is restarted.
+* If *both* CPU and memory resources are changed simultaneously, the container is restarted (due to the memory policy).
 
 ## Limitations
 
-In-place resize of pod resources currently has the following limitations:
+For Kubernetes {{< skew currentVersion >}}, in-place resource resizing has the following limitations:
```

Comment:
suggest changing `in-place resource resizing` -> `in-place pod resize` in order to be consistent on what we want to call this feature.

Reply:
Flipped it around to avoid naming the feature instead.

Reply:
Looks good! thanks